model/task/task.go

```diff
@@ -1264,6 +1252,12 @@ func SetTasksScheduledTime(tasks []Task, scheduledTime time.Time) error {
 	return nil
 }
 
+// GetTaskIdBetweenIds retrieves the task id between two tasks that are of the same.
+func GetTaskIdBetweenTasks(t1, t2 Task) (*Task, error) {
```

Comment:
We should call this `FindTaskBetweenTasks` (or `FindMidwayTask` or something) since this is getting the whole task data model rather than just its ID.